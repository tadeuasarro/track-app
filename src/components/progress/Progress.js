import Footer from '../footer/Footer';

const Progress = () => (
  <div>
    <Footer />
  </div>
);

export default Progress;
