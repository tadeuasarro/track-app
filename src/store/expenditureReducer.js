import { SET_EXPENDITURES } from '../actions/expenditure';
import filterExpenditures from '../helpers/filterExpenditures';

const initialState = {
  expenditures: [],
  summary: false,
};

const expenditureReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_EXPENDITURES:
      return {
        expenditures: action.payload,
        summary: filterExpenditures(action.payload),
      };
    default:
      return state;
  }
};

export default expenditureReducer;
