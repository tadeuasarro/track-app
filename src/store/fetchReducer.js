import { SET_LOADING, SET_ERRORS } from '../actions/fetch';

const initialState = {
  loading: false,
  errors: false,
};

const fetchReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_LOADING:
      return {
        loading: true,
        errors: false,
      };
    case SET_ERRORS:
      return {
        loading: false,
        errors: action.payload,
      };
    default:
      return state;
  }
};

export default fetchReducer;
