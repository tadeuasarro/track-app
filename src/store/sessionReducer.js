import { SET_CURRENT_USER, UPDATE_USER_TARGET } from '../actions/session';

const initialState = {
  username: false,
  id: false,
  target: false,
  token: false,
};

const sessionReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_CURRENT_USER:
      return {
        username: action.payload.user.username,
        id: action.payload.user.id,
        target: action.payload.user.target,
        token: action.payload.token,
      };
    case UPDATE_USER_TARGET:
      return {
        ...state,
        target: action.payload,
      };
    default:
      return state;
  }
};

export default sessionReducer;
