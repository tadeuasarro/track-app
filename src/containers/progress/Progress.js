import { useSelector } from 'react-redux';
import Expenditure from '../expenditure/Expenditure';
import Loading from '../../components/loading/Loading';
import Footer from '../../components/footer/Footer';
import './progress.css';

const Progress = () => {
  const { expenditure } = useSelector(state => state);

  const { expenditures, summary } = expenditure.expenditures;

  if (!expenditure.expenditures) return <Loading />;

  return (
    <div>
      <div className="progress-container">
        <div className="progress-summary">
          <div className="progress-expent">
            <h3>Total expent </h3>
            <h3 className="total-amount">
              {(summary.total / 1000).toFixed(1)}
              &nbsp;
              k
            </h3>
          </div>
        </div>
        <div className="history-container">
          <i className="fas fa-history" />
          <h3>History</h3>
          { expenditures.map(expenditure => (
            <Expenditure key={expenditure.id} expenditure={expenditure} />)
          )}
        </div>
      </div>
      <Footer />
    </div>
  );
};

export default Progress;
