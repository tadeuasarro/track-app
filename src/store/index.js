import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import sessionReducer from './sessionReducer';
import expenditureReducer from './expenditureReducer';
import fetchReducer from './fetchReducer';

const store = createStore(
  combineReducers({
    session: sessionReducer,
    expenditure: expenditureReducer,
    fetch: fetchReducer,
  }),
  applyMiddleware(thunk),
);

export default store;
