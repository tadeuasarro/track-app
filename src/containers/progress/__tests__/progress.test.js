import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../store/index';
import Progress from '../Progress';

describe('Rendering component', () => {
  it('creates a Progress component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Progress />
          </Provider>
        </BrowserRouter>,
      );
    });
    // eslint-disable-next-line
    const element = screen.getByText('History');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const comp = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Progress />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(comp).toMatchSnapshot();
  });
});
