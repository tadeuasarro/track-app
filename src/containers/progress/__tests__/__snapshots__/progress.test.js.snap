// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div>
  <div
    className="progress-container"
  >
    <div
      className="progress-summary"
    >
      <div
        className="progress-expent"
      >
        <h3>
          Total expent 
        </h3>
        <h3
          className="total-amount"
        >
          NaN
          Â  k
        </h3>
      </div>
    </div>
    <div
      className="history-container"
    >
      <i
        className="fas fa-history"
      />
      <h3>
        History
      </h3>
    </div>
  </div>
  <div
    className="footer-container"
  >
    <a
      className="footer-link"
      href="/measures"
      onClick={[Function]}
    >
      <i
        className="fas fa-balance-scale-right"
      />
    </a>
    <a
      className="footer-link"
      href="/"
      onClick={[Function]}
    >
      <i
        className="fas fa-bars"
      />
    </a>
    <a
      className="footer-link"
      href="/progress"
      onClick={[Function]}
    >
      <i
        className="fas fa-tasks"
      />
    </a>
    <a
      className="footer-link"
      href="/more"
      onClick={[Function]}
    >
      <i
        className="fas fa-info"
      />
    </a>
  </div>
</div>
`;
